merkledag.go

```diff
@@ -0,0 +1,64 @@
+package format
+
+import (
+	"context"
+	"fmt"
+
+	cid "github.com/ipfs/go-cid"
+)
+
+var ErrNotFound = fmt.Errorf("merkledag: not found")
+
+// Either a node or an error.
+type NodeOption struct {
+	Node Node
+	Err  error
+}
+
+// The basic Node resolution service.
+type NodeGetter interface {
+	Get(context.Context, *cid.Cid) (Node, error)
+
+	// TODO(ipfs/go-ipfs#4009): Remove this method after fixing.
+	OfflineNodeGetter() NodeGetter
+}
+
+// NodeGetters can optionally implement this interface to make finding linked
+// objects faster.
+type LinkGetter interface {
+	NodeGetter
+	// TODO(ipfs/go-ipld-format#9): This should return []*cid.Cid
+	GetLinks(ctx context.Context, nd *cid.Cid) ([]*Link, error)
```

Comment:
Yes; the idea is to avoid requiring that *all* DAGServices implement this method (even it they don't cache links).